DEVELOPER_GUIDE.md

```diff
@@ -89,6 +96,120 @@ If you are retrospectively adding the API, Then you can refer to the following r
 - [OpenSearch Documentation](https://opensearch.org/docs/latest)
 - [Go through the serialisation logic in OpenSearch](https://github.com/opensearch-project/OpenSearch)
 
+## Adding a test-case for API definition
+
+Once you've finished with the model API, follow the steps below to create a test-case.
+
+### Pre-requisites
+1. In the *build.gradle.kts* file, include the dependency. 
+
+   ```
+    classpath("software.amazon.smithy:smithy-openapi:1.21.0")
+   ```
+  
+2. In the *smithy-build.json* file, include the plugin. 
+
+    ```
+     "plugins": {
+          "openapi": {
+              "service": "OpenSearch#OpenSearch",
+              "protocol": "aws.protocols#restJson1"
+          }
+      }
+    ```   
+
```

Comment:
could we permanently add these to smithy-build instead of adding when we have to add test cases?

Reply:
Yes, corrected in latest commit [91c26ab](https://github.com/opensearch-project/opensearch-api-specification/pull/45/commits/91c26abb5b2aad6b02351e9e6e9453b10a4e1524).